controllers/space/space_controller.go

```diff
@@ -257,65 +257,60 @@ func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1a
 
 // listSpaceBindings searches all the SpaceBindings of a given Space,
 // it will use the name of the parentSpace for the research if there is one.
-func (r *Reconciler) listSpaceBindings(space *toolchainv1alpha1.Space) (toolchainv1alpha1.SpaceBindingList, error) {
-	// list SpaceBindings bound to the space
-	spaceBindings := toolchainv1alpha1.SpaceBindingList{}
+func (r *Reconciler) listSpaceBindings(space *toolchainv1alpha1.Space, foundBindings []toolchainv1alpha1.SpaceBinding) ([]toolchainv1alpha1.SpaceBinding, error) {
+	parentBindings := toolchainv1alpha1.SpaceBindingList{}
 	if err := r.Client.List(context.TODO(),
-		&spaceBindings,
+		&parentBindings,
 		runtimeclient.InNamespace(space.Namespace),
 		runtimeclient.MatchingLabels{
 			toolchainv1alpha1.SpaceBindingSpaceLabelKey: space.Name, // use current space name for the search
 		},
 	); err != nil {
-		// return the error in case of read issues
-		return spaceBindings, err
+		// return child bindings and the error in case of read issues
+		return foundBindings, err
 	}
 
-	// if there's no parentSpace set let's return the SpaceBindings from above
+	// spaceBindings is the list that will be returned, it will contain either parent and child merged or just the "parent" ones retrieved above.
+	spaceBindings := mergeSpaceBindings(foundBindings, parentBindings.Items)
```

Comment:
I would either reuse the foundBindings variable
```go
	foundBindings := mergeSpaceBindings(foundBindings, parentBindings.Items)
```
or use name `mergedBindings`
```go
	mergedBindings := mergeSpaceBindings(foundBindings, parentBindings.Items)
```
it will make the reading of the rest of the function easier